### ISSUE TITLE
Make `.chezmoi.configFile` template variable unavailable or empty if no config file is found

### ISSUE THREAD

**Issue by felipecrs** [twpayne/chezmoi#1722]
## Is your feature request related to a problem? Please describe.

I have a script which performs a migration from `chezmoi.toml` to `chezmoi.yaml`:

https://github.com/felipecrs/dotfiles/blob/9730e3943658cbd2064f3e1d7c7baff960d4a364/home/run_always_before_0_migrate.sh.tmpl

However, when there are no configuration files, `.chezmoi.configFile` still points at the default `chezmoi.toml`, which makes my script think that it's there.

I think this is misleading.

## Describe the solution you'd like

The `.chezmoi.configFile` could be set only when chezmoi actually read the configuration from that file.

## Describe alternatives you've considered

```diff
- # {{ if not (hasSuffix ".yaml" .chezmoi.configFile) }}
+ # {{ if and (stat .chezmoi.configFile) (not (hasSuffix ".yaml" .chezmoi.configFile)) }}
```

## Additional context

None.

**Comment by felipecrs** [comment]
The workaround is completely ok by the way, but I just wanted to report this so at least we can discuss if this is working as intended or not.

**Comment by twpayne** [comment]
I think the current behavior is reasonable and the work-around is sufficient.

**Comment by felipecrs** [comment]
> I think the current behavior is reasonable

The docs say:

> ![image](https://user-images.githubusercontent.com/29582865/145687793-cba64d56-e0d5-4100-a9b3-633ae7ce991c.png)

_if any_ to mean would mean if there is any configuration file being used/present.

> the work-around is sufficient.

Yeah. It is.

**Comment by twpayne** [comment]
Thanks, docs fixed with a556b93d19c12b41d46eab1d363131a460290575.
